app/assets/javascripts/components/nav/nav.jsx

```diff
@@ -141,6 +144,7 @@ const Nav = () => {
                     <li>
                       <b><a href={`/users/${encodeURIComponent(currentUser)}`} className="current-user">{currentUser}</a></b>
                     </li>
+                    <Provider store={store} ><NewsHandler/></Provider>
```

Comment:
I don't like the idea of adding a separate store for use by the Nav component. It would be better to do one of the following:
* modify app.jsx to pass the same store to both the Nav and Main components
* embed the news data in the HTML so that it can be extracted and rendered without using a store
* follow the pattern of NotificationsBell to request and render the news data within the component lifecycle, without using a store.

(I also noticed that the `createStore` function that we use in create_store.js is deprecated, but that's probably outside the scope of this PR.)

Reply:
Thank you for the feedback. I will make the required changes.

Reply:
> (I also noticed that the `createStore` function that we use in create_store.js is deprecated, but that's probably outside the scope of this PR.)

I'll be happy to address the createStore issue after resolving all the news feature-related problems. If no one has taken up the createStore issue by then.